p2p/router.go

```diff
@@ -350,6 +350,15 @@ func (r *Router) AddChannelDescriptors(chDescs []*ChannelDescriptor) {
 	for _, chDesc := range chDescs {
 		r.chDescs = append(r.chDescs, *chDesc)
 	}
+	// FIXME: We need to pass the channel descriptors on to the underlying
+	// transport. Here we assume an mconn transport but in the future we need to
+	// find a better way to do this. Most likely by providing the channel
+	// descriptors on initialization of the transport
```

Comment:
is this the first time we've put strong dependency in the router on mconn? Would it be enough to add a "getChDescs" method on the transport interface that we could call?

Reply:
I would actually prefer that

Reply:
I wasn't sure if we wanted to make the transport agnostic to the channel descriptors and instead provide them directly to the connections